appinventor/blocklyeditor/src/blocks/utilities.js

```diff
@@ -75,6 +75,29 @@ Blockly.Blocks.Utilities.wrapSentence = function(str, len) {
   }
 };
 
+// Change the text of collapsed blocks on rename
+// Recurse to fix collapsed parents
+
+Blockly.Blocks.Utilities.MAX_COLLAPSE = 4;
```

Comment:
@shrutirij Why the max of 4 and what happens when the recursion depth is greater?